signer/src/storage/postgres.rs

```diff
@@ -12,6 +52,111 @@ impl PgStore {
     pub async fn connect(url: &str) -> Result<Self, sqlx::Error> {
         Ok(Self(sqlx::PgPool::connect(url).await?))
     }
+
+    /// Write parts of the Stacks Nakamoto block headers to the database.
+    async fn write_stacks_block_header(&self, blocks: &[NakamotoBlock]) -> Result<(), Error> {
+        let block_summaries: Vec<StacksBlockSummary> = blocks
+            .iter()
+            .map(|block| StacksBlockSummary {
+                block_id: block.block_id().to_hex(),
+                chain_length: block.header.chain_length as i64,
+                parent_block_id: block.header.parent_block_id.to_hex(),
+            })
+            .collect();
+
+        if block_summaries.is_empty() {
+            return Ok(());
+        }
+
+        sqlx::query(
```

Comment:
Nit: I prefer using the macros, which makes it quicker and easier for me to verify that the queries are correct.

```suggestion
        sqlx::query!(
```